x-pack/plugins/index_management/public/application/components/component_templates/component_template_details/component_template_details.tsx

```diff
@@ -0,0 +1,152 @@
+  EuiSpacer,
+  EuiCallOut,
+} from '@elastic/eui';
+
+import { SectionLoading, TabSettings, TabAliases, TabMappings } from '../shared_imports';
+import { useComponentTemplatesContext } from '../component_templates_context';
+import { TabSummary } from './tab_summary';
+import { ComponentTemplateTabs, TabType, Tab } from './tabs';
+import { ManageButton, ManageAction } from './manage_button';
+
+interface Props {
+  componentTemplateName: string;
+  onClose: () => void;
+  showFooter?: boolean;
+  actions?: ManageAction[];
+}
+
+export const ComponentTemplateDetailsFlyout: React.FunctionComponent<Props> = ({
+  componentTemplateName,
+  onClose,
+  actions,
+}) => {
+  const { api } = useComponentTemplatesContext();
+
+  const { data: componentTemplateDetails, isLoading, error } = api.useLoadComponentTemplate(
+    componentTemplateName
+  );
+
+  const [activeTab, setActiveTab] = useState<Tab>(TabType.Summary);
+
+  let content: React.ReactNode | undefined;
+
+  if (isLoading) {
+    content = (
+      <SectionLoading>
+        <FormattedMessage
+          id="xpack.idxMgmt.componentTemplateDetails.loadingIndexTemplateDescription"
+          defaultMessage="Loading component template…"
+        />
+      </SectionLoading>
+    );
+  } else if (error) {
+    content = (
+      <EuiCallOut
+        title={
+          <FormattedMessage
+            id="xpack.idxMgmt.componentTemplateDetails.loadingErrorMessage"
+            defaultMessage="Error loading component template"
+          />
+        }
+        color="danger"
+        iconType="alert"
+        data-test-subj="sectionError"
+      >
+        <p>{error.message}</p>
+      </EuiCallOut>
+    );
+  } else if (componentTemplateDetails) {
+    const {
+      template: { settings, mappings, aliases },
+    } = componentTemplateDetails;
+
+    const tabToComponentMap: {
+      [key: string]: React.ReactNode;
+    } = {
+      [TabType.Summary]: <TabSummary componentTemplateDetails={componentTemplateDetails} />,
+      [TabType.Settings]: <TabSettings settings={settings} />,
+      [TabType.Mappings]: <TabMappings mappings={mappings} />,
+      [TabType.Aliases]: <TabAliases aliases={aliases} />,
+    };
+
+    const tabContent = tabToComponentMap[activeTab];
+
+    content = (
+      <>
+        <ComponentTemplateTabs activeTab={activeTab} setActiveTab={setActiveTab} />
+
+        <EuiSpacer size="l" />
+
+        {tabContent}
+      </>
+    );
+  }
+
+  return (
+    <EuiFlyout
+      onClose={onClose}
+      data-test-subj="componentTemplateDetails"
+      aria-labelledby="componentTemplateDetailsFlyoutTitle"
+      size="m"
+      maxWidth={500}
+    >
+      <EuiFlyoutHeader>
+        <EuiTitle size="m">
+          <h2 id="componentTemplateDetailsFlyoutTitle" data-test-subj="title">
+            {componentTemplateName}
+          </h2>
+        </EuiTitle>
+      </EuiFlyoutHeader>
+
+      <EuiFlyoutBody data-test-subj="content">{content}</EuiFlyoutBody>
+
+      {actions && (
```

Comment:
Maybe we sill want to be able to close the flyout even though we donn't provide actions?

Reply:
I'm not sure it's necessary since there's already the ability to close it via the `x` icon at the top right of the flyout.